services/provider/client/client.go

```diff
@@ -2,49 +2,50 @@ package client
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	pb "github.com/red-hat-storage/ocs-operator/services/provider/pb"
 	"google.golang.org/grpc"
-	"k8s.io/klog"
 )
 
-type ConsumerClient struct {
+type OCSProviderClient struct {
 	Client  pb.OCSProviderClient
 	timeout time.Duration
 }
 
-// NewConsumerClient creates a ConsumerClient to talk to OCS consumer server
-func NewConsumerClient(cc *grpc.ClientConn, timeout time.Duration) *ConsumerClient {
-	return &ConsumerClient{Client: pb.NewOCSProviderClient(cc), timeout: timeout}
+// NewProviderClient creates a client to talk to OCS provider server
+func NewProviderClient(cc *grpc.ClientConn, timeout time.Duration) *OCSProviderClient {
+	return &OCSProviderClient{Client: pb.NewOCSProviderClient(cc), timeout: timeout}
 }
 
 // NewGRPCConnection returns a grpc client connection which can be used to create the consumer client
 // Note: Close the connection after use
 func NewGRPCConnection(serverAddr string, opts []grpc.DialOption) (*grpc.ClientConn, error) {
 	conn, err := grpc.Dial(serverAddr, opts...)
 	if err != nil {
-		klog.Fatalf("failed to dial: %v", err)
+		return nil, fmt.Errorf("failed to dial: %v", err)
 	}
 	return conn, err
 }
 
-// OnBoardConsumer to validate the consumer and create StorageConsumer
+// OnboardConsumer to validate the consumer and create StorageConsumer
 // resource on the StorageProvider cluster
-func (cc *ConsumerClient) OnBoardConsumer(token, capacity string) (*pb.OnBoardConsumerResponse, error) {
-	req := &pb.OnBoardConsumerRequest{
-		Token:    token,
-		Capacity: capacity,
+func (cc *OCSProviderClient) OnboardConsumer(token, name, capacity string) (*pb.OnboardConsumerResponse, error) {
+	req := &pb.OnboardConsumerRequest{
+		Token:        token,
+		ConsumerName: name,
+		Capacity:     capacity,
 	}
 
 	ctx, cancel := context.WithTimeout(context.Background(), cc.timeout)
```

Comment:
Why are we creating a new context.Background() for each call? It will be preferable if all calls during a single reconcile run use the same context. This is true for the entire OCS operator reconcile run. Not only the new part.

Reply:
This is from the mock PR

Reply:
Ok